pytypes/type_util.py

```diff
@@ -406,12 +410,12 @@ def _deep_type(obj, checked, checked_len, depth = None, max_sample = None):
     # a common fake length of checked list:
     checked_len2 = len(checked)
     if res == tuple:
-        res = Tuple[tuple(_deep_type(t, checked, checked_len2, depth-1) for t in obj)]
+        res = Tuple[tuple(_deep_type(t, checked, checked_len2, depth-1, get_type=get_type) for t in obj)]
```

Comment:
I usually avoid using keyword args in heavily used internal functions because of higher overhead. ``get_type=get_type`` is not even shorter than
``None, get_type``. Also not significantly better readable thanks to the expressive name ``get_type``.
CPython constructs a dictionary for every such call, which have more overhead than a tuple. IIRC it reuses tuples much better than dictionaries.

Reply:
Yes, runtime typechecking is not about performance, but this is a trivial improvement.

Reply:
Updated.